test/test_ops.py

```diff
@@ -461,6 +461,28 @@ def test_nms_cuda_float16(self):
         keep16 = ops.nms(boxes.to(torch.float16), scores.to(torch.float16), iou_thres)
         self.assertTrue(torch.all(torch.eq(keep32, keep16)))
 
+    def test_batched_nms_implementations(self):
+        """Make sure that both implementations of batched_nms yield identical results"""
+
+        num_boxes = 1000
+        iou_threshold = .9
+
+        boxes = torch.cat((torch.rand(num_boxes, 2), torch.rand(num_boxes, 2) + 10), dim=1)
+        assert max(boxes[:, 0]) < min(boxes[:, 2])  # x1 < x2
+        assert max(boxes[:, 1]) < min(boxes[:, 3])  # y1 < y2
```

Comment:
nit: it might be preferable to construct boxes which are always well-formed.
So something like
```python
boxes = torch.rand(num_boxes, 4)
boxes[:, 2:] += boxes[:, :2]
```
is generally better when constructing bounding boxes. It might be good to move this to a helper function btw